File: q2/best_script.py
'''
--[[
    created by : Narate Ketram
    twitter @eNarate
    facebook https://www.facebook.com/koonnarate
--]]
'''
fin = open('numbers.txt', 'r')
fout = open('run_result.txt', 'w')
din = []
for line in fin.readlines():
    din.append(line.strip())
    
dout = {}
for i in din:
    dout[i] = 0

for k in din:
    dout[k] = dout[k] + 1
    
for k, v in dout.items():
    if v > 1:
        fout.write("\"" + str(k) + "\", " + str(v) + "\n")
fin.close()
fout.close()
print "done :)"

File: q1/best_script.lua
--[[
    created by : Narate Ketram
    twitter @eNarate
    facebook https://www.facebook.com/koonnarate
--]]
local fin = io.open('numbers.txt', 'r')
local fout = io.open('run_result.txt', 'w')
local din = {}

if fin then
    for line in fin:lines() do
        if #line ~= 0 then
            table.insert(din,line)
        end
    end
    table.sort(din)
    local count, before, current, diff = 0, -1, 0, 0
    for i,current in ipairs(din) do
        diff = current - before
        while diff > 1 do
            fout:write(string.format("%04d\n", count));
            diff = diff - 1
            count = count + 1
        end
        before = current
        count = count + 1
    end
    fin:close()
    fout:close()
end
print('done :)')
